include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -729,6 +755,16 @@ class __future : private std::tuple<_Args...>
         return __storage.__wait_and_get_value(__my_event);
     }
 
+    constexpr auto
+    __wait_and_get_value(const std::shared_ptr<__result_and_scratch_storage_base>& __p_storage)
```

Comment:
I propose to rewrite this overload:
```C++
__wait_and_get_value(__result_and_scratch_storage_base* __p_result_and_scratch_storage_base)
```

Reply:
I think there is at least two reasons why it's the bad pattern to use const reference for shared pointer for overloading:
1) Semantically this means nothing because we using reference to shared-pointer: we don't owns the object on reference counter, managed by this shared-pointer.
2) As described at https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr, we have non-explicit
`constexpr shared_ptr( std::nullptr_t ) noexcept;`
So why we should call this overload for `nullptr` parameter value it's not obvious at all.

Reply:
What I propose to change: please take a look to https://github.com/uxlfoundation/oneDPL/pull/2032
@danhoeflinger please take a look too.

Reply:
OK